src/Offer/ReadModel/JSONLD/OfferLDProjector.php

```diff
@@ -418,19 +418,19 @@ protected function applyImageRemoved(AbstractImageRemoved $imageRemoved): ?JsonD
         );
 
         // Unset the main image if it matches the removed image
-        // @see https://jira.uitdatabank.be/browse/III-4684
-        // some historic ImageRemoved Events have http instead of https in the eventStore
-        if (isset($offerLd->image) && stristr($offerLd->{'image'}, ':') === stristr($imageUrl, ':')) {
+        if (isset($offerLd->image) && $offerLd->{'image'} === $imageUrl) {
             unset($offerLd->{'image'});
         }
 
         if (!isset($offerLd->image) && count($filteredMediaObjects) > 0) {
             $offerLd->image = array_values($filteredMediaObjects)[0]->contentUrl;
         }
 
-        // If no media objects are left remove the attribute.
+        // If no media objects are left remove the attribute and the imageUrl.
+        // @see III-4684
```

Comment:
I think this is a sensible think to do, if there is no `mediaObject` property it never hurts to also unset the `image` just in case it's still present 👍 

However, I'd also keep the initial check that you added because the PR now doesn't cover this scenario for example:

- Event has 2 images, at least one imported from UDB2 using `http://`
- The image with `http://` is the main image
- The image with `http://` is removed

➡️ The removed image with `http://` will still be set as the main image if I'm not mistaken.

But this is also a very niche edge case given that XML imports are no longer possible. For existing places that were imported from UDB2 it might be relevant, but still a minor issue. I'd personally re-add the `stristr` check (and keep this extra `unset()`), but all in all your `unset()` when `mediaObject` is empty will also cover _most_ cases.

Reply:
@JonasVHG maybe best to also add unit test for this edge case?